src/SignalR/server/Core/src/Internal/HubMethodDescriptor.cs

```diff
@@ -27,10 +34,28 @@ public HubMethodDescriptor(ObjectMethodExecutor methodExecutor, IEnumerable<IAut
                 ? MethodExecutor.AsyncResultType
                 : MethodExecutor.MethodReturnType;
 
-            if (IsChannelType(NonAsyncReturnType, out var channelItemType))
+            foreach (var closedType in NonAsyncReturnType.AllBaseTypes())
             {
-                IsChannel = true;
-                StreamReturnType = channelItemType;
+                if (!closedType.IsGenericType)
+                {
+                    continue;
+                }
+
+                var openType = closedType.GetGenericTypeDefinition();
+
+                if (openType == typeof(IAsyncEnumerable<>))
```

Comment:
Does this ever work? AllBaseTypes works for `ChannelReader<>` because it is inherited from. `IAsyncEnumerable<>` is implemented.

Reply:
Good catch. The previous iteration worked for hub methods that returned IAsyncEnumerable directly (like generators), but not for hub methods that return implementing types.